presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/TestSphericalGeoFunctions.java

```diff
@@ -204,6 +205,140 @@ private void assertArea(String wkt, double expectedArea)
+        assertContains("10 10, 10 30, 20 30, 20 10, 10 10", "20 -7.9", false);
+
+        // other external points
+        assertContains("10 10, 10 30, 20 30, 20 10", "50 41", false);
+        assertContains("10 10, 10 30, 20 30, 20 10, 10 10", "15 40", false);
+        assertContains("10 10, 10 30, 20 30, 20 10, 10 10", "5 20", false);
+        assertContains("10 10, 10 30, 20 30, 20 10, 10 10", "25 20", false);
+        assertContains("10 10, 10 30, 20 30, 20 10, 10 10", "15 2", false);
+        assertContains("10 -10, 10 -30, 20 -30, 25 -8, 10 -10", "15 -2", false);
+        assertContains("-10 10, -10 30, -20 30, -25 8, -10 10", "-15 35", false);
+
+        // points inside and outside a triagle
+        assertContains("-50 70, -15 65, -30 50", "-45 68", true);
+        assertContains("-50 70, -15 65, -30 50", "-50 20", false);
+        assertContains("-50 70, -15 65, -30 50", "-50 70", false);
+        assertContains("-50 70, -15 65, -30 50", "-15 65", false);
+        assertContains("-50 70, -15 65, -30 50", "-30 50", false);
+        assertContains("-50 70, -15 65, -30 50", "-30 41.8", false);
+        assertContains("-50 70, -15 65, -30 50", "-46.2 60.777", false);
+        assertContains("-50 70, -15 65, -30 50", "-50 59.23", false);
+
+        // based on flight routes from salt lake city -> buffalo -> ft. lauderdale -> salt lake city
+        assertContains(saltBuffFort, wichita, true);
+        assertContains(saltBuffFort, grandRapids, true);
+        assertContains(saltBuffFort, fortLauderdale, false);
+        assertContains(saltBuffFort, newYork, false);
+        assertContains(saltBuffFort, santiago, false);
+        assertContains(saltBuffFort, oklahomaCity, false);
+        assertContains(saltBuffFort, buffalo, false);
+
+        // based on flight routes from LA -> melbourne -> sydney -> SF -> LA
+        assertContains(losAngMelSydSanFran, sydney, false);
+        assertContains(losAngMelSydSanFran, southeastOfSydney, true);
+        assertContains(losAngMelSydSanFran, noumea, false);
+        assertContains(losAngMelSydSanFran, honolulu, false);
+        assertContains(losAngMelSydSanFran, grandRapids, false);
+
+        // polygon that looks like "I", starting from lower left corner
+        assertContains("30 -10, 30 5, 35 5, 35 15, 30 15, 30 20, 50 20, 50 15, 45 15, 45 5, 50 5, 50 -10", "40 11", true);
+        assertContains("30 -10, 30 5, 35 5, 35 15, 30 15, 30 20, 50 20, 50 15, 45 15, 45 5, 50 5, 50 -10", "47 14", false);
+        assertContains("30 -10, 30 5, 35 5, 35 15, 30 15, 30 20, 50 20, 50 15, 45 15, 45 5, 50 5, 50 -10", "27 25", false);
+
+        // polygon with a hole, starting from lower left corner
+        assertContains("-10 -80, -30 -80, -30 -50, -10 -50), (-12 -72, -20 -72, -20 -12", "-18 -70", false);
+        assertContains("-10 -80, -30 -80, -30 -50, -10 -50), (-12 -72, -20 -72, -20 -12", "-25 -55", true);
+    }
+
+    private String buildPolygon(String[] polygonPoints)
```

Comment:
this method can be simplified to

```
    private String buildPolygon(String...points)
    {
        return Arrays.stream(points).collect(Collectors.joining(", "));
    }
```

Then, the callers can be simplified as well:

```
        String saltBuffFort = buildPolygon(saltLakeCity, buffalo, fortLauderdale, saltLakeCity);
        String losAngMelSydSanFran = buildPolygon(losAngeles, melbourne, sydney, sanFrancisco);
```